qcodes/utils/helpers.py

```diff
@@ -19,19 +18,41 @@
 
 _tprint_times= {} # type: Dict[str, float]
 
+
 log = logging.getLogger(__name__)
 
+
 class NumpyJSONEncoder(json.JSONEncoder):
-    """Return numpy types as standard types."""
-    # http://stackoverflow.com/questions/27050108/convert-numpy-type-to-python
-    # http://stackoverflow.com/questions/9452775/converting-numpy-dtypes-to-native-python-types/11389998#11389998
+    """
+    This JSON encoder adds support for serializing types that the built-in
+    `json` module does not support out-of-the-box. See the docstring of the
+    `default` method for the description of all conversions.
```

Comment:
For future reference I think we should add a link to the cpython issue that you referenced. There may come a day where this is fixed upstream

Reply:
resolving that might not help with supporting __all__ the numpy types though. so, i'm not sure that linking adds value.

Reply:
I still think that it's not intuitive why we need to convert numbers and documenting that is likely to save some time but it should probably be documented along with that specific part i.e. just after https://docs.scipy.org/doc/numpy/reference/arrays.scalars.html

Reply:
agreed offline to add more information to the PR's description.